### ISSUE TITLE
Allow synchronous continuations in certain scenarios

### ISSUE THREAD

**Issue by eyalpost** [StackExchange/StackExchange.Redis#85]
The current discussion about the timeout issue reminded me a feature request I had, which is the ability to have synchronous continuations. I would basically define this as "I know what I'm doing. Let me have full control".
There are some situations (mainly when developing a class which wraps SE.Redis) that a synchronous cont. would make a lot of sense. For example, imagine I have the following method:

``` C#
Task<DateTime> GetValueAsDate(string key)
{
 //pseudo code
  return SE.Redis.Get(key).continueWith(value=>ConvertToDate(value));
}
```

so basically the code above would fetch the data from Redis and convert it to Date before passing back to the caller. We still want to maintain the async flow so we're returning a task but since the conversion to Date is negligible, we don't want to use a thread from the pool just for that. If the caller is also adding a continuation then today, for that continuation to run, there will be another thread pool scheduling.

What do you think?

**Comment by NickCraver** [comment]
@eyalpost ping for updated thoughts here

**Comment by eyalpost** [comment]
@mgravell  can you give an example of how to implement such a continuation?
My feeling was that for the above example, there would be an additional thread taken from the pool which would perform the ConvertToDate() function and then will be disposed. Seems wasteful to take a threadpool thread just for that

**Comment by mgravell** [comment]
> My feeling was that for the above example, there would be an additional thread taken from the pool which would perform the ConvertToDate() function and then will be disposed. Seems wasteful to take a threadpool thread just for that

that's a perfect scenario for a pool thread; pool threads aren't "disposed" at the end of running something - so that should be fine?

**Comment by mgravell** [comment]
We have not specified `RunContinuationsAsynchronously` for quite some time, and we have fundamentally changed the threading model around continuations in 2.0 - basically : it should behave much more like you expect in 2.0; considering closed re https://github.com/StackExchange/StackExchange.Redis/issues/871
